lib/vu_state.go

```diff
@@ -26,6 +26,14 @@ type TracerProvider interface {
 	Tracer(name string, options ...trace.TracerOption) trace.Tracer
 }
 
+// AddrResolver is an interface for DNS resolution.
+type AddrResolver interface {
+	// ResolveAddr looks up the IP address for the given host and optionally port.
+	// The address is expected in the form "host:port" or just "host".
+	// It returns the resolved IP, the port (0 if not specified), and an error if any.
+	ResolveAddr(addr string) (net.IP, error)
```

Comment:
I would say that it follows the settings from the options and maybe list those. Or at least a note it does it in a special way that mgiht nto be just what `dig hostname` will return on the cli